### PR TITLE
manifest parser

### PR DESCRIPTION
**Pull request by emileten**
added a parser to automate the reading of intermediary output file paths in argo manifests. Notes : 

1. This function is to be used [here](https://github.com/ClimateImpactLab/downscaleCMIP6/blob/master/notebooks/downscaling_pipeline/global_validation_manual.ipynb). It will typically be run to build the `data_dict` object, to avoid having to manual find and write these paths. It should be run like :  

`get_output_paths(manifest: dict, regex: str)` where `regex` could be something like `'(?=.*create-coarse-reference)(?=.*move-chunks-to-space)'`. I have to write the calls to this function when I'll have identified the current, stable template names. 

2. This function assumes the argo manifest is already read in and converted to a `dict` object. We still need to find the way to obtain the argo manifest from the code. 

@dgergel I forked this repo because I don't have write access. Also for some reasons I can't assign you the review !

### CODE REVIEW

**Comment by dgergel** [comment]
@emileten this looks good, thanks! I think we'll have an updated stable template name soon, sorry for the confusion with the multiple templates. 

We can add the argo manifest read-in and conversion in a separate PR since you already have a concise PR description written up. And then add the calls to the function at the same time.

File: notebooks/downscaling_pipeline/science_validation_manual.py [hunk 1]
```diff
@@ -178,3 +179,41 @@ def read_gcs_zarr(zarr_url, token='/opt/gcsfuse_tokens/impactlab-data.json', che
...
+def get_output_paths(manifest, regex):
+    """
+    lists status.nodes in an argo manifest, and grabs intermediary output files paths using the node tree represented by
+    status.nodes[*].name. Keeps only nodes of type 'Pod' and phase 'succeeded'.
+
+    Parameters
+    ----------
+    manifest : dict
+    regex : str
+        regular expression syntax str to filter nodes based on which templates where executed within a given node and before that given
```
Comment (dgergel) [inline notebooks/downscaling_pipeline/science_validation_manual.py:191]: where --> were

**Review by dgergel — CHANGES_REQUESTED** [review]
I just noticed one typo - otherwise this is good to go.

File: notebooks/downscaling_pipeline/science_validation_manual.py (hunk 1 above)
Comment (emileten) [reply notebooks/downscaling_pipeline/science_validation_manual.py]: thanks ! fixed

**Comment by emileten** [comment]
@dgergel would you mind merging this ? I think I don't have write access.

**Pull request merged**